soccer/planning/planner/LineKickPlanner.cpp

```diff
@@ -0,0 +1,204 @@
+    if (brutePath) return std::move(*brutePath);
+
+    Point targetVel = (aimTarget - ball.pos).normalized(*_approachSpeed);
+    Point targetPos = ball.pos - targetVel.normalized(Robot_Radius * 3);
+
+    request.dynamic_obstacles.push_back(ball.dynamicObs());
+    RobotInstant targetInstant{Pose{targetPos, 0}, Twist{targetVel, 0},
+                               RJ::Time{0s}};
+    request.motionCommand = PathTargetCommand{targetInstant};
+    Trajectory path = _pathTargetPlanner.plan(std::move(request));
+    path.setDebugText("Gives ups");
+    _targetKickPos = std::nullopt;
+    PlanAngles(path, startInstant, AngleFns::facePoint(aimTarget),
+               rotationConstraints);
+    return std::move(path);
+}
+
+Trajectory LineKickPlanner::planForSlowMovingBall(PlanRequest&& request) {
+    constexpr double ballAvoidDistance = 0.05;
+    Point aimTarget = std::get<LineKickCommand>(request.motionCommand).target;
+    const Ball& ball = request.context->state.ball;
+    Trajectory& prevTrajectory = request.prevTrajectory;
+    RobotInstant startInstant = request.start;
+    RotationConstraints rotationConstraints = request.constraints.rot;
+    Point targetVel = (aimTarget - ball.pos).normalized(*_approachSpeed);
+    Point targetPos;
+    if (std::abs(targetVel.angleBetween(
+            ball.pos - startInstant.pose.position())) > 10 * M_PI / 180.0) {
+        targetPos = ball.pos -
+                    targetVel.normalized(ballAvoidDistance * 2 + Robot_Radius);
+        if (!prevTrajectory.empty() &&
+            targetPos.distTo(prevTrajectory.last().pose.position()) <
+                PathTargetPlanner::goalPosChangeThreshold() &&
+            _reusePathCount < 20) {
+            targetVel = prevTrajectory.last().velocity.linear();
+            _reusePathCount++;
+        } else {
+            _reusePathCount = 0;
+        }
+        request.static_obstacles.add(
+            std::make_shared<Circle>(ball.pos, ballAvoidDistance));
+    } else {
+        targetPos = ball.pos + targetVel.normalized(Robot_Radius);
+        if (!prevTrajectory.empty() &&
+            ball.pos.distTo(prevTrajectory.last().pose.position()) <
+                PathTargetPlanner::goalPosChangeThreshold() &&
+            _reusePathCount < 20) {
+            targetVel = prevTrajectory.last().velocity.linear();
+            _reusePathCount++;
+        } else {
+            _reusePathCount = 0;
+        }
+    }
+    _targetKickPos = std::nullopt;
+    RobotInstant targetInstant{Pose{targetPos, 0}, Twist{targetVel, 0},
+                               RJ::Time{0s}};
+    request.motionCommand = PathTargetCommand{targetInstant};
+    Trajectory path = _pathTargetPlanner.plan(std::move(request));
+    PlanAngles(path, startInstant, AngleFns::facePoint(aimTarget),
+               rotationConstraints);
+    path.setDebugText("Slow Moving Ball");
+    return std::move(path);
+}
+
+std::optional<Trajectory> LineKickPlanner::attemptBruteForce(
```

Comment:
This brute-force logic is close to common between the planners, right? Could we somehow combine it?

(Also, comments pls)